### ISSUE TITLE
Flaky tests

### ISSUE THREAD

**Issue by rvagg** [nodejs/gyp-next#42]
We're getting this pretty regularly in node-gyp CI: https://travis-ci.com/github/nodejs/node-gyp/jobs/339883554

```
The command "npm test" exited with 0.
10714.02s$ GYP_MSVS_VERSION=2015 GYP_MSVS_OVERRIDE_PATH="C:\\Dummy" python -m pytest
1072============================= test session starts ==============================
1073platform linux -- Python 3.5.6, pytest-4.6.6, py-1.7.0, pluggy-0.13.1
1074rootdir: /home/travis/build/nodejs/node-gyp
1075collected 27 items                                                             
1076
1077gyp/pylib/gyp/MSVSSettings_test.py F.F.....                              [ 29%]
...
```

I don't really know how to interpret the log output that follows, but it's been consistent. I can restart it in Travis and it should pass again, this shows up in maybe 50% of Travis runs and it seems to be most common on Node.js 12 + Python 3.5, but I don't have numbers to back that up.

**Comment by ryzokuken** [comment]
Thanks for reporting this. @targos can you reproduce this fail locally?

**Comment by targos** [comment]
It's probably the same problem as https://github.com/nodejs/gyp-next/pull/7

**Comment by targos** [comment]
I don't have python 3.5 to test locally

**Comment by ryzokuken** [comment]
Good catch, it _is_ infact that problem. It'd be a shame to not support Python 3.5. I'd try to look into it in a week or so.

**Comment by cclauss** [comment]
Python 3.5 EOLs in September https://devguide.python.org/#status-of-python-branches

**Comment by cclauss** [comment]
Can this issue be closed?
